src/Symfony/Component/Serializer/Normalizer/ConstraintViolationListNormalizer.php

```diff
@@ -32,8 +34,12 @@ class ConstraintViolationListNormalizer implements NormalizerInterface, Cacheabl
     private $defaultContext;
     private $nameConverter;
 
-    public function __construct($defaultContext = [], NameConverterInterface $nameConverter = null)
+    public function __construct(/** ContextInterface */ $defaultContext = null, NameConverterInterface $nameConverter = null)
     {
+        if(!$defaultContext instanceof ContextInterface && is_array($defaultContext)){
+            $defaultContext = new DefaultContext($defaultContext);
+            @trigger_error("you must pass an object of type ". ContextInterface::class .", the passage of an array is depreciated", E_USER_DEPRECATED);
+        }
```

Comment:
@weaverryan  @dunglas is this what you had in mind? We (me and @flug) are working actively on this, we changed only one normalizer for now and wanted to check the implementation with you before going forward.

Reply:
No it's not. As I've described in the other comments, I think we should keep an array or an iterable here. Using an interface will make the component hard to extend, and it will be a pain to migrate the current API Platform context options to this new mechanism. All other Symfony components use plain arrays for this kind of things, for consistency, and because it's easier to extend, I think we should keep the array but just provide an helper function to build this context instead.